common/lib/common-definitions/src/events.ts

```diff
@@ -15,13 +15,24 @@ export interface IErrorEvent extends IEvent {
 }
 
 export interface IEventProvider<TEvent extends IEvent> {
+    /**
+     * Registers a callback to be invoked when the corresponding event is triggered.
+     */
     readonly on: IEventTransformer<this, TEvent>;
+
+    /**
+     * Registers a callback to be invoked the first time the corresponding event is triggered.
```

Comment:
Not sure how else to word this but `first time` seems misleading.  It more like `invoked once when the next time `?

Reply:
That's fair. Perhaps "first time after registration"?

Reply:
```suggestion
     * Registers a callback to be invoked the first time (after registration) the corresponding event is triggered.
```